src/content/actions/types.js

```diff
@@ -76,7 +76,8 @@ type TimelineAction =
    { type: 'CHANGE_TIMELINE_EXPANDED_THREAD', threadIndex: ThreadIndex, isExpanded: boolean };
 
 type URLEnhancerAction =
-  { type: "@@urlenhancer/urlSetupDone" };
+  { type: "@@urlenhancer/urlSetupDone" } |
+  { type: '@@urlenhancer/updateURLState', urlState: any };
```

Comment:
I haven't dug much into the urlState. I wonder if at some point (in the future) we could strictly type it.